setup_routing.go

```diff
@@ -22,10 +23,32 @@ import (
 	"github.com/libp2p/go-libp2p/core/network"
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/libp2p/go-libp2p/core/routing"
+	prometheus "github.com/prometheus/client_golang/prometheus"
 	"go.opencensus.io/stats/view"
 	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
 )
 
+func init() {
+	promRegistry, ok := prometheus.DefaultRegisterer.(*prometheus.Registry)
+	if !ok {
+		goLog.Error("routing metrics: error casting DefaultRegisterer")
+		return
+	}
+	pe, err := ocprom.NewExporter(ocprom.Options{
+		Namespace: "ipfs",
+		Registry:  promRegistry,
+		OnError: func(err error) {
+			goLog.Errorf("ocprom error: %w", err)
+		},
+	})
+	if err != nil {
+		goLog.Errorf("routing metrics: error creating exporter: %w", err)
+		return
+	}
+	view.RegisterExporter(pe)
+	view.SetReportingPeriod(2 * time.Second)
```

Comment:
- [ ] check if we need 2s resolution, i think Prometheus Scrape Interval is ~15s

Reply:
our grafana is polling every 30s, so 2s is overkill for prod, but it may be useful for local debugging.

let's keep 2s to match Kubo, we can adjust in future PR if 2s turns out to be too costly